### ISSUE TITLE
All metadata strictly either/or in posts.json or frontmatter

### ISSUE THREAD

**Issue by dreeves** [beeminder/blog#379]
```[tasklist]
### Desiderata
- [x] Break the build if any field for any post is specified in both posts.json and in the frontmatter of the source doc (even if the values for the field match)
```

(Originally this was "fail loudly if slugs don't match" but we generalized it.)

## Replicata

1. Write a post
2. Specify "slug": "foo" in the posts.json
3. Specify "slug: bar" in the YAML frontmatter

## Expectata

Break the build.

## Resultata

One of them silently overrides the other (I forget which).

## Cognata

* #318 
* #380 
* #382
* #377
* #452 (PR FOR THIS)

_**Verbata**: inconsistent slugs, failing loudly,_

**Comment by dreeves** [comment]
Related topics of discussion:

1. Do we want to change the frontmatter format from YAML to JSON?
2. Do we even want frontmatter in the source docs, vs putting everything in posts.json?

**Comment by narthur** [comment]
@dreeves Yup. I feel like it would be best to have answers for those questions before I resolve this issue, since otherwise I may be spending time implementing a check we don't need.

**Comment by dreeves** [comment]
I don't have strong opinions on either of those, other than just having a clear aesthetic preference for JSON over YAML.

If it were possible to force _all_ metadata into the source doc frontmatter I might like that, but that's too hard with the legacy posts currently. (Maybe in the future when we have API access to etherpads, or whatever we replace etherpads with, then we can do that migration.)

So right now our choices are between having consistency by killing frontmatter (only a few posts using it so far!) or the status quo where you can mix-and-match where the metadata lives.

Here's a consideration: We need the title in the source doc, because doc.bmndr.co needs it for the previews. So if all metadata lives in posts.json then we have to unDRYly have the title in two places and remember to keep them in sync when edit the title.

I guess that's pushing me towards metadata in the source doc a bit. Unless title is a special case? No, that's dumb. Let's say we stick with mix-and-match and say that metadata in posts.json (other than slug?) is deprecated.

Do you have any thoughts from the dev perspective?

**Comment by narthur** [comment]
@dreeves This seems reasonable to me. So:

- Fail builds if metadata is added to posts.json on non-legacy posts

And... that's it?

**Comment by dreeves** [comment]
What defines legacy vs non? Date threshold? Explicit field?

**Comment by narthur** [comment]
@dreeves I've been planning to use a date threshold.

**Comment by dreeves** [comment]
I think we're now finding ways to avoid needing to distinguish legacy-vs-non. See discussion in https://github.com/beeminder/blog/issues/380#issuecomment-1758382829 and the new main desideratum for this gissue, namely, break the build if you try to specify any metadata redundantly.

**Comment by narthur** [comment]
This should be resolved with #452
